src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImpl.java

```diff
@@ -1245,4 +1249,177 @@ public SBApiResponse workflowBulkUpdateTransitionV1(String userAuthToken, Multip
+			Map<String, Object> mdoInfo = (Map<String, Object>) mdoDetails.get(mdoUserId);
+			String departmentName = (String) mdoInfo.get(Constants.ROOT_ORG_NAME);
+			String rootOrgId = (String) mdoInfo.get(Constants.ROOT_ORG_ID);
+			Map<String, Object> allPendingRequest = new HashMap<>();
+			int limit = configuration.getPendingRequestCountLimit();
+			List<WfStatusEntity> pendingRequestsEntityList = wfStatusRepo.getListOfApplicationUsingDept(Constants.PROFILE_SERVICE_NAME, Constants.SEND_FOR_APPROVAL, departmentName, limit);
+			if(CollectionUtils.isEmpty(pendingRequestsEntityList)) {
+				String message = "There are no approval requests pending for approval with the MDO";
+				InputStream inputStream = new ByteArrayInputStream(message.getBytes());
+				InputStreamResource inputStreamResource = new InputStreamResource(inputStream);
+				return ResponseEntity.status(HttpStatus.OK).body(inputStreamResource);
+			}
+			userIdSet.clear();
+			for (WfStatusEntity wfStatusEntity : pendingRequestsEntityList) {
+				userIdSet.add(wfStatusEntity.getUserId());
+				ObjectMapper objectMapper = new ObjectMapper();
+				List<Map<String, Object>> valuesToBeUpdate = objectMapper.readValue(wfStatusEntity.getUpdateFieldValues(), new TypeReference<List<Map<String, Object>>>() {
+				});
+				List<Map<String, Object>> pendingRequestList;
+				for(Map<String, Object> valueToUpdate : valuesToBeUpdate){
+					if(valueToUpdate.containsKey(Constants.TO_VALUE)){
+						Map<String, Object> updateKeyValue = (Map<String, Object>) valueToUpdate.get(Constants.TO_VALUE);
+						String keyToUpdate = updateKeyValue.keySet().stream().findFirst().get();
+						if(Constants.DESIGNATION.equalsIgnoreCase(keyToUpdate) || Constants.GROUP.equalsIgnoreCase(keyToUpdate)){
+							if(allPendingRequest.containsKey(wfStatusEntity.getUserId())) {
+								pendingRequestList = (List<Map<String, Object>>) allPendingRequest.get(wfStatusEntity.getUserId());
+								pendingRequestList.add(updateKeyValue);
+								allPendingRequest.put(wfStatusEntity.getUserId(), pendingRequestList);
+							} else {
+								pendingRequestList = new ArrayList<>();
+								pendingRequestList.add(updateKeyValue);
+								allPendingRequest.put(wfStatusEntity.getUserId(), pendingRequestList);
+							}
+						}
+					}
+				}
+			}
+			String csvFilePath = "/tmp/pendingRequest.csv";
+			Map<String, Object> allUserDetails = this.getUserSearchDetails(userIdSet,false, rootOrgId);
+			if(MapUtils.isNotEmpty(allUserDetails)){
+				this.populateSheetWithPendingRequests(allPendingRequest, allUserDetails ,csvFilePath);
+			} else {
+				String message = "There are no requests pending for Group or Designation approval with the MDO";
+				InputStream inputStream = new ByteArrayInputStream(message.getBytes());
+				InputStreamResource inputStreamResource = new InputStreamResource(inputStream);
+				return ResponseEntity.status(HttpStatus.OK).body(inputStreamResource);
```

Comment:
This must be internal error message -- it is due to failing to enrich the user details... 
update the same accordingly... 
if pendingRequestsEntityList is empty -- then it won't come here...
so pendingRequestsEntityList is not empty and userIdSet is also not empty...

Reply:
This is done as suggested by @karthik-tarento